python/ray/data/_internal/execution/operators/map_transformer.py

```diff
@@ -59,6 +61,9 @@ def input_type(self) -> MapTransformFnDataType:
     def output_type(self) -> MapTransformFnDataType:
         return self._output_type
 
+    def set_target_max_block_size(self, target_max_block_size: int):
+        self._target_max_block_size = target_max_block_size
```

Comment:
Prefer to set it here, since right now all map operators have a generic map transformer, not necessarily one that ends in BuildOutputBlocksMapTransformFn.

Reply:
oh, what I mean is `MapTransformer` and `BuildOutputBlocksMapTransformFn` will have `set_target_max_block_size`, but not other MapTransformFns.
Then map op calls `map_transformer. set_target_max_block_size`. and in `MapTransformer.__call__`, only call `set_target_max_block_size` for `BuildOutputBlocksMapTransformFn`.
Not a huge deal though, I'll leave the decision to you.
By the way, I think eventually we should make every  `MapTransformer` end with `BuildOutputBlocksMapTransformFn`.

Reply:
Yeah the problem is that not all MapTransformers will end with a BuildOutputBlocksMapTransformFn, so either we set it for all MapTransformFns or we check explicitly if it's a BuildOutputBlocksMapTransformFn. The former seemed easier to me right now, but yeah later we could make them all end with BuildOutputBlocksMapTransformFn.